oak-blob-cloud/src/test/java/org/apache/jackrabbit/oak/blob/cloud/s3/TestS3Ds.java

```diff
@@ -94,6 +115,66 @@ public void setUp() throws Exception {
         super.setUp();
     }
 
+    @Test
+    public void testInitiateDirectUploadUnlimitedURIs() throws DataRecordUploadException,
+            RepositoryException {
+        ConfigurableDataRecordAccessProvider ds
+                  = (ConfigurableDataRecordAccessProvider) createDataStore();
+        long uploadSize = ONE_GB * 50;
+        int expectedNumURIs = 5000;
+        DataRecordUpload upload = ds.initiateDataRecordUpload(uploadSize, -1);
+        Assert.assertEquals(expectedNumURIs, upload.getUploadURIs().size());
+
+        uploadSize = ONE_GB * 100;
+        expectedNumURIs = 10000;
+        upload = ds.initiateDataRecordUpload(uploadSize, -1);
+        Assert.assertEquals(expectedNumURIs, upload.getUploadURIs().size());
+
+        uploadSize = ONE_GB * 200;
+        upload = ds.initiateDataRecordUpload(uploadSize, -1);
+        Assert.assertEquals(expectedNumURIs, upload.getUploadURIs().size());
+    }
+
+    @Test
+    public void testGetDownloadURIIT() throws IOException, RepositoryException {
+        DataRecord record = null;
+        DataRecordAccessProvider ds = (DataRecordAccessProvider) createDataStore();
+        InputStream testStream = randomStream(0, 256);
+        record = doSynchronousAddRecord(ds, testStream);
+        URI uri = ((DataRecordAccessProvider) ds).getDownloadURI(record.getIdentifier(), DataRecordDownloadOptions.DEFAULT);
+        Assert.assertNotNull("uri is null", uri);
+    }
+
+    @Test
+    public void testCompleteDirectUploadUpdate() throws DataRecordUploadException, RepositoryException {
+        DataRecord uploadedRecord = null;
+        ConfigurableDataRecordAccessProvider ds
+        = (ConfigurableDataRecordAccessProvider) createDataStore();
+        DataRecordUpload uploadContext = ds.initiateDataRecordUpload(ONE_MB, 1);
+
+        //Pull the blob id out and modify it
```

Comment:
Not sure about this ...but is there a need for all this computation below? See test case https://github.com/apache/jackrabbit-oak/blob/trunk/oak-jcr/src/test/java/org/apache/jackrabbit/oak/jcr/binary/BinaryAccessTest.java#L218-L245